File: week_02/src/main/java/Question16.java
/*
Autor  = Zaid Nayeb
Date   = 18/11/2025

16.	Print numbers from 1 to 100
Description: Use a loop to print num\bers sequentially.
Example:
Output:
1, 2, 3, …, 100

*/

public class Question16 {
    public static void main(String[] args) {
        System.out.println(printNumbers());
    }
    public static String printNumbers(){
        StringBuilder sb = new StringBuilder();
        for (int i = 1; i <= 100; i++){
            sb.append(i);
            if (i < 100){
                sb.append(", ");
            }
        }
        return sb.toString();
    }
}

File: week_03/src/Question_16.js
// Author : Zaid Nayeb
// Date   : 2025/11/28

/*
16.	Fibonacci sequence 
Description: Print the Fibonacci sequence up to N terms. 
Example: 
Input: 
7 
Output: 
0, 1, 1, 2, 3, 5, 8 


*/
console.log(fibonacci(7));

function fibonacci(num){
    let arr = [0,1];
    let left = 0;
    let right = 1;
    while (arr[arr.length - 1] <= num) {
        arr.push( arr[left] + arr[right]);
        left++;
        right++;
    }
    return arr;
}
module.exports = fibonacci;
